addons/l10n_dk_oioubl/models/account_edi_xml_oioubl_201.py

```diff
@@ -0,0 +1,248 @@
+    'direct debit': 49,
+    'compensation': 97,
+}
+UBL_TO_OIOUBL_TAX_CATEGORY_ID_MAPPING = {
+    # Simple mapping between tax type provided in UBL and what is accepted in OIOUBL
+    # https://docs.peppol.eu/poacc/billing/3.0/codelist/UNCL5305/
+    # https://www.oioubl.info/codelists/en/urn_oioubl_id_taxcategoryid-1.1.html
+    'AE': 'ReverseCharge',
+    'E': 'ZeroRated',
+    'S': 'StandardRated',
+    'Z': 'ZeroRated',
+    'G': 'ZeroRated',
+    'O': 'ZeroRated',
+    'K': 'ReverseCharge',
+    'L': 'ZeroRated',
+    'M': 'ZeroRated',
+}
+TAX_POSSIBLE_VALUES = set(UBL_TO_OIOUBL_TAX_CATEGORY_ID_MAPPING.values())
+
+
+def format_vat_number(partner):
+    vat = partner.vat.replace(' ', '')
+    if vat[:2].isnumeric():
+        vat = partner.country_code.upper() + vat
+    return vat
+
+
+class AccountEdiXmlOIOUBL201(models.AbstractModel):
+    _name = "account.edi.xml.oioubl_201"
+    _inherit = 'account.edi.xml.ubl_20'
+    _description = "OIOUBL 2.01"
+
+    # Data validation Schematron available at the following URL:
+    # https://rep.erst.dk/git/openebusiness/common/-/tree/master/resources/Schematrons/OIOUBL
+
+    # -------------------------------------------------------------------------
+    # EXPORT
+    # -------------------------------------------------------------------------
+
+    def _export_invoice_filename(self, invoice):
+        return f"{invoice.name.replace('/', '_')}_oioubl_201.xml"
+
+    def _export_invoice_ecosio_schematrons(self):
+        return {
+            'invoice': 'org.oasis-open:invoice:2.0',
+            'credit_note': 'org.oasis-open:creditnote:2.0',
+        }
+
+    def _export_invoice_vals(self, invoice):
+        vals = super()._export_invoice_vals(invoice)
+        vals['vals'].update({
+            'customization_id': 'OIOUBL-2.01',
+            'AddressType_template': 'l10n_dk_oioubl.oioubl_201_AddressType',
+            # ProfileID is the property that define which documents the company can send and receive
+            # 'Procurement-BilSim-1.0' is the simplest one: invoice and bill
+            # https://www.oioubl.info/documents/en/en/Guidelines/OIOUBL_GUIDE_PROFILES.pdf
+            'profile_id': 'Procurement-BilSim-1.0',
+            'profile_id_attrs': {
+                'schemeID': 'urn:oioubl:id:profileid-1.6',
+                'schemeAgencyID': DANISH_NATIONAL_IT_AND_TELECOM_AGENCY_ID,
+            }
+        })
+        vals['vals'].setdefault('invoice_type_code_attrs', {}).update({
+            'listID': 'urn:oioubl:codelist:invoicetypecode-1.2',
+            'listAgencyID': DANISH_NATIONAL_IT_AND_TELECOM_AGENCY_ID,
+        })
+
+        return vals
+
+    def _get_partner_party_vals(self, partner, role):
+        vals = super()._get_partner_party_vals(partner, role)
+        endpoint_id = format_vat_number(partner)
+        if endpoint_id[:2] == 'DK':
+            scheme_id = 'DK:CVR'
+        elif endpoint_id[:2] == 'FR':
+            scheme_id = 'FR:SIRET'
```

Comment:
???

Reply:
!!!

Reply:
If you have a question to ask, ask the question. "???" doesn't mean anything.
Obviously, French can't do the same as other and just use their VAT number

Reply:
And if you are asking for the Danish, they want danish to use their CVR number.

See, I don't even know what you don't understand, hence why you should ask your question.